### PR TITLE
Latest Guests Component

### LINKED ISSUE #15: Latest Guests Component
image

### PR DESCRIPTION
**Pull request by ayse8888**
Hello All,
I built the latest guest component #15 
From the mock data that I created, I mapped the items and displayed in the UI.
In the desktop version, there is a view button on the pictures. I created a modal when clicked. we don't know the data structure yet but i thought it may work like this. We can remove it if it is not necessary though.
Additionally, for the mobile version, the UI will change and a slider will be replaced ( until screen size max 1024px ). I used "carousel" component from Bootstrap for this.
I attach my visuals here as well:

image

image

image

### CODE REVIEW

**Comment by Rofy87** [comment]
Hi @ayse8888 
The code seems good to be 
I liked what you have done with the UI and the carousel

**Review by mustafaguldag — APPROVED** [review]
Hi Ayşe, code seems nice and I liked the design and slider. Thank you for your contribute!

File: src/components/LatestGuests/GuestItems.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
+import React, { useState } from "react";
+import { Col, Modal, Button } from "react-bootstrap";
+import "./LatestGuests.css";
+
+const GuestItems = ({ img, buttonText, petName, petAge, petBreed, id }) => {
+  const [show, setShow] = useState(false);
+
+  const handleClose = () => setShow(false);
```
Comment (halitbatur) [inline src/components/LatestGuests/GuestItems.js]: this function and the following one can be done in 1 function which take param in order to change the setshow

File: src/components/LatestGuests/LatestGuests.css [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+  padding: 2rem 1rem 1rem 0 !important;
+  font-size: 25px;
+}
+
+#modalFooter {
+  background: url("https://t4.ftcdn.net/jpg/01/06/77/37/360_F_106773743_CdPrTCQabRIeXPjCZddr3qV1PGT34rn7.jpg");
+  border-top: none !important;
+}
+
+/* Carousel CSS, those classes are coming from bootstrap data, not showing in html.
```
Comment (halitbatur) [inline src/components/LatestGuests/LatestGuests.css:73]: good comment

File: src/components/LatestGuests/LatestGuests.css [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+.carousel-indicators li {
+  width: 10px !important;
+  height: 10px !important;
+  border-radius: 100% !important;
+  background-color: #ff7777 !important;
+}
+
+/* 
+for screens under 1024px, cards will not be displayed,
```
Comment (halitbatur) [inline src/components/LatestGuests/LatestGuests.css]: unconsistant commet style /* should be at the begining of the line

**Review by halitbatur — CHANGES_REQUESTED** [review]

File: src/components/LatestGuests/GuestItems.js (hunk 1 above)
Comment (ayse8888) [reply src/components/LatestGuests/GuestItems.js]: created 1 function which handles the setShow and it is working on the modal for show and close

File: src/components/LatestGuests/LatestGuests.css (hunk 2 above)
Comment (ayse8888) [reply src/components/LatestGuests/LatestGuests.css:73]: thank you !

File: src/components/LatestGuests/LatestGuests.css (hunk 3 above)
Comment (ayse8888) [reply src/components/LatestGuests/LatestGuests.css]: done

File: src/components/LatestGuests/GuestContainer.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+                    petAge={guest.petAge}
+                    petBreed={guest.petBreed}
+                  />
+                ))}
+              </Row>
+            </>
+          )}
+        </Col>
+      </Row>
+
```
Comment (halitbatur) [inline src/components/LatestGuests/GuestContainer.js]: extra new line here

File: src/components/LatestGuests/GuestItems.js [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+        <Button
+          className="shadow-none"
+          id="viewButton"
+          variant="primary"
+          onClick={() => handleToggle()}
+        >
+          {buttonText}
+        </Button>
+      </Col>
+
```
Comment (halitbatur) [inline src/components/LatestGuests/GuestItems.js]: extra new line

File: src/components/LatestGuests/guestcontainer.test.js [hunk 6]
```diff
@@ -0,0 +1,11 @@
+import React from "react";
+import renderer from "react-test-renderer";
+import GuestContainer from "./GuestContainer";
+import { latestAddedGuests } from "./data";
+
+it("renders correctly", () => {
```
Comment (halitbatur) [inline src/components/LatestGuests/guestcontainer.test.js]: can we change the test name to descripe which component is getting used maybe like render Guest Container correctly

File: src/components/LatestGuests/guestitems.test.js [hunk 7]
```diff
@@ -0,0 +1,8 @@
+import React from "react";
+import renderer from "react-test-renderer";
+import GuestItems from "./GuestItems";
+
+it("renders correctly", () => {
```
Comment (halitbatur) [inline src/components/LatestGuests/guestitems.test.js]: test name should be better

File: src/components/LatestGuests/GuestContainer.js (hunk 4 above)
Comment (ayse8888) [reply src/components/LatestGuests/GuestContainer.js]: done

File: src/components/LatestGuests/GuestItems.js (hunk 5 above)
Comment (ayse8888) [reply src/components/LatestGuests/GuestItems.js]: done

File: src/components/LatestGuests/guestcontainer.test.js (hunk 6 above)
Comment (ayse8888) [reply src/components/LatestGuests/guestcontainer.test.js]: done

File: src/components/LatestGuests/guestitems.test.js (hunk 7 above)
Comment (ayse8888) [reply src/components/LatestGuests/guestitems.test.js]: done

**Review by halitbatur — APPROVED** [review]

**Pull request merged**